internal/controllers/clusterrequest_controller.go

```diff
@@ -235,17 +235,16 @@ func (t *clusterRequestReconcilerTask) run(ctx context.Context) (ctrl.Result, er
 		}
 	}
 
-	crProvisionedCond := meta.FindStatusCondition(
-		t.object.Status.Conditions, string(utils.CRconditionTypes.ClusterProvisioned))
-	if crProvisionedCond != nil {
```

Comment:
I think we should keep this check to ensure the ordering and avoid setting the configurationApplied condition when hw provisioning is still in-progress and no ClusterInstance has been created (no ClusterProvisioned status condition)

Reply:
We are setting the condition to status `False` with a `ClusterNotReady` reason. @edcdavid and I discussed that it's nice to know the configuration before it's actually being applied, especially since all the policies are in enforce.

Reply:
Yeah, I saw the ClusterNotReady condition you added. I think it makes sense when the cluster has started provisioning, so we have conditions like HWProvisioned, ClusterProvisioned-InProgress, and ConfigurationApplied-NotReady.

However, it feels a bit weird to me that when the ClusterInstance hasn’t been created yet because HW provisioning is still in progress, we will have the conditions HWProvisioned-InProgress and ConfigurationApplied=ClusterNotReady. This doesn’t align with how other conditions are handled, where we only add the condition when the previous step is completed.

Reply:
Let's just keep `if crProvisionedCond != nil`  then and not check for the condition reason.